### PR TITLE
Fix crash when player is in a special team

### PR DESCRIPTION
**Pull request by LightBlueCube**
QQ图片20240905070404

my English is bad, im sorry for that :(

### CODE REVIEW

File: Northstar.CustomServers/mod/scripts/vscripts/mp/_gamestate_mp.nut [hunk 1]
```diff
@@ -1062,7 +1062,9 @@ void function OnPlayerChangedTeam( entity player )
 	if ( !player.hasConnected ) // Prevents players who just joined to trigger below code, as server always pre setups their teams
 		return
 	
-	NotifyClientsOfTeamChange( player, GetOtherTeam( player.GetTeam() ), player.GetTeam() )
+	// only TEAM_IMC and TEAM_MILITIA can use function GetOtherTeam(), doesnt need to notify them when player got a special team
+	if( [ TEAM_IMC, TEAM_MILITIA ].contains( player.GetTeam() ) )
```
Comment (Zanieon) [inline Northstar.CustomServers/mod/scripts/vscripts/mp/_gamestate_mp.nut:1066]:
```suggestion
	if( IsIMCOrMilitiaTeam( player.GetTeam() ) )
```
Utility function that exactly serves this purpose.

**Review by Zanieon — CHANGES_REQUESTED** [review]

**Review by Zanieon — APPROVED** [review]
Code looks good, and this will probably solve problems in regards to any gamemode isn't 2 sided based, or even if they partially uses Militia or IMC teams, it will work only between those teams.

**Comment by GeckoEidechse** [comment]
Uh, what's the command to switch player teams again so that I can test this? ^^"

**Comment by Zanieon** [comment]
> Uh, what's the command to switch player teams again so that I can test this? ^^"

There's no default way of doing it right now, you can only do either by using FSU and typing `!switch` in chat, or using my FD branch and typing `changeteam` in console.

**Comment by F1F7Y** [comment]
You could also do smth like `script SetTeam(GetPlayerArray()[0], GetOtherTeam(GetPlayerArray()[0].GetTeam()))` ( <- this prob dont work as i dont remember the exact funcs )

**Comment by LightBlueCube** [comment]
> Uh, what's the command to switch player teams again so that I can test this? ^^"

u can try use this function
```
void function SetTeam( entity ent, int team )
```
just like this comment

> You could also do smth like `script SetTeam(GetPlayerArray()[0], GetOtherTeam(GetPlayerArray()[0].GetTeam()))` ( <- this prob dont work as i dont remember the exact funcs )

here is some team name u may need to know
TEAM_ANY, TEAM_INVALID, TEAM_UNASSIGNED, TEAM_SPECTATOR, TEAM_IMC, TEAM_MILITIA, TEAM_BOTH

**Comment by Zanieon** [comment]
Tested this by using #872 and playing Free For All, so far server didn't complain, and probably it wouldn't as there's sanity checks in that team changing code.

Besides... Only FFA Scoreboard supports more than 2 teams, all other gamemodes will crash clients that tries to open Scoreboard if trying to assign players to other than `TEAM_MILITIA` or `TEAM_IMC` teams

I also have to mention that i prefer this to get merged first over #875 so i can update the function there manually as Github probably will get confused and do a merge conflict for either PRs that gets merged last.

**Comment by GeckoEidechse** [comment]
Merging based on reviews, thanks @Zanieon

**Pull request merged**